### ISSUE TITLE
Broken licensing conditions

### ISSUE THREAD

**Issue by juliangilbey** [zaquestion/lab#294]
Hello,

Thanks for this project, which we are planning to package for Debian.

I see that you chose to use Unlicensed to license this project.  Unfortunately, that's a very problematic license that seems not to have been written with the consultation of lawyers, and therefore almost certainly doesn't do what you think it does.  See the discussion thread on the license-review mailing list starting at [http://lists.opensource.org/pipermail/license-review_lists.opensource.org/2012-January/001360.html](http://lists.opensource.org/pipermail/license-review_lists.opensource.org/2012-January/001360.html) which explains the issues with that license in detail.

If you would like to use a public-domain license, my strong recommendation (as discussed on that thread) is to use the Creative Commons CC0 license instead, found at [https://creativecommons.org/share-your-work/public-domain/](https://creativecommons.org/share-your-work/public-domain/).  This license was created (with legal input) to do what you intend as best as is currently legally possible.  There is a tool on that page ("Use this tool") for creating the correct wording for your project.  It is also (afaik) an internationally recognised license.

Best wishes!

**Comment by zaquestion** [comment]
@juliangilbey Thanks for the input, using The Unlicense was a known risk w.r.t. the complexities of making something public domain whilst also licensing it. I'll take a look at the materials and likely take your suggestion to change licenses.


I use Debian as my daily driver, so I deeply support getting this packaged over there. Admittedly the debian packaging process is still a bit much for me to grok given the amount of time I've got. Regardless, please let me know how I can help.

We use GoReleaser to manage our builds which support outputting `.deb` files if that helps https://goreleaser.com/nfpm/

**Comment by juliangilbey** [comment]
@zaquestion Hi Zaq? - pleasure!  A preliminary Debian package was made by Felix Lechner, and I'm sponsoring it for him.

There are two small issues that we've patched locally - would you prefer me to send them to you via email, via a bug report or via a merge request?

Finally, the current version of the package failed to build for me.  I'll report that as a separate issue.

Best wishes,

Julian

**Comment by zaquestion** [comment]
@juliangilbey pull request to GitHub works best, though I'd rather get them than not, so whichever means achieve that :D

**Comment by anarcat** [comment]
any progress on fixing the license here? i'd also love to have this in debian and might be able to help if this is unblocked...

**Comment by zaquestion** [comment]
Yeah I'd like to switch over to the CC0 license. I haven't had a chance to go through license creator tool, but if someone wants to take a swing at it and put up an MR to change licenses, thats fine with me.

**Comment by zaquestion** [comment]
@juliangilbey @anarcat --  @ianfoo has put up changes to move us over to the Creative Commons license. I've brought them in, can you do a last spot check and LMK if looks up to snuff?

**Comment by juliangilbey** [comment]
Hi @zaquestion and @ianfoo - thanks for doing this!  It looks mostly fine to me, as it seems to be copied and pasted from CC.

Two things I noticed which could be tweaked:

(1) In the README.md file, add:

```
This work is published from:
<span property="vcard:Country" datatype="dct:ISO3166"
      content="US" about="https://github.com/zaquestion/lab">
  United States</span>.
```

before the closing `</p>`.  (This is what CC recommends.)

(2) In the LICENSE file, remove the paragraph from lines 5-13 ("CREATIVE COMMONS CORPORATION IS NOT A LAW FIRM..."), as that is a communication from CC to you, not from you to the reader/user of your software.

Best wishes,

Julian

**Comment by zaquestion** [comment]
@juliangilbey I've updated with your suggestions, thanks for the help here apologies for dropping the ball. Assuming you'll want a new tag with this in it, I'll try to prioritize that for soon, I need to address an issue with project creation that came in after some gitlab.com changes https://github.com/zaquestion/lab/issues/329

**Comment by anarcat** [comment]
so licensing is fixed here? maybe the issue can be closed then, or should that wait for a release?

in any case, thanks for taking those questions seriously, it's much appreciated!

**Comment by dod38fr** [comment]
Well, `goreleaser.yml` still nentions `unlicense`:
```
$ ack -i unlicense
.goreleaser.yml
33:  license: Unlicense
```

**Comment by dod38fr** [comment]
On the other hand, the main license file was switched to CC0 and released with version 0.16.0 .

**Comment by zaquestion** [comment]
@dod38fr Thanks for the heads up, it's fixed on master now. I searched for any more mentions and didn't see any. Thanks for helping get this closed out.
